src/SoftDeleteable/Mapping/Event/Adapter/ORM.php

```diff
@@ -44,7 +44,7 @@ public function getDateValue($meta, $field)
 
         return $this->getObjectManager()->getConnection()->convertToPHPValue(
             $this->getRawDateValue($mapping),
-            $mapping instanceof FieldMapping ? $mapping->type : ($mapping['type'] ?? Types::DATETIME_MUTABLE)
+            $mapping instanceof FieldMapping ? $mapping->type : ($mapping->type ?? $mapping['type'] ?? Types::DATETIME_MUTABLE)
```

Comment:
There's already an instanceof check, so either that or the null coalescing stays, but we don't need both.